### DISCUSSION TITLE
Feature request: platform-specific override for default-filter configuration

### DISCUSSION THREAD

**Discussion by nagisa** [nextest-rs/nextest#1869] [Feature requests]
### Motivation

I want to ignore a family of tests that I know does not run on a certain platform by default.

I tried 

```
[[profile.default.overrides]]
platform = 'cfg(target_os = "linux")'
default-filter = "none()"
```

and have found that this is not taken into account.

### Proposal

I propose that platform-specific overrides for `default-filter` should be applied.

### Alternatives

filterset could be extended to allow applying rules based on the platform? That would make it conditional, but...

I have also considered `#[cfg_attr(..., ignore)]` but in my case the tests affected are easier to filter out on a granularity of an entire crate/package, rather than individual tests.

Since I use a `Justfile`, I also have attempted to figure something out with the `-E` flags, but I have found that then it makes it difficult for users to specify their own filtersets via this flag as multiple `-E` flags are `union`ed rather than subset as is the behaviour with the `default-filter` set.

**Comment by sunshowers** [comment]
Yes -- this was definitely something I was intending to do as a followup!

**Comment by sunshowers** [comment]
We're going to likely need it at work soon, so implemented this: https://github.com/nextest-rs/nextest/pull/1871

**Comment by sunshowers** [comment]
All right, this is part of cargo-nextest 0.9.84: https://nexte.st/docs/running/#per-platform-default-filters. Would love to hear your thoughts!
